modules/java-modules/http/src/main/java/org/syslog_ng/http/HTTPDestination.java

```diff
@@ -85,9 +81,12 @@ public boolean isOpened() {
 
 
     @Override
-    public boolean send(String message) {
+    public boolean send(LogMessage msg) {
         int responseCode = 0;
+        String message;
         try {
+            this.url = new URL(options.getURL().getResolvedString(msg, getTemplateOptionsHandle(), LogTemplate.LTZ_SEND));
```

Comment:
As the URL itself is derived from the actual message(and URL is not used only in this method...) it should be a local variable.